apps/lib/apps.c

```diff
@@ -2106,29 +2106,40 @@ unsigned char *next_protos_parse(size_t *outlen, const char *in)
     return out;
 }
 
-void print_cert_checks(BIO *bio, X509 *x,
+int check_cert(BIO *bio, X509 *x,
                        const char *checkhost,
-                       const char *checkemail, const char *checkip)
+                       const char *checkemail, const char *checkip, int print)
 {
+    int valid_host = 0;
+    int valid_mail = 0;
+    int valid_ip = 0;
+    int ret = 1;
+
     if (x == NULL)
-        return;
+        return 0;
+
     if (checkhost) {
```

Comment:
When you're touching the code, could you please change all the occurrences of pointer to boolean conversion in the function to `pointer != NULL` ?